proposals/0185-vote-account-v4.md

```diff
@@ -0,0 +1,243 @@
+
+Vote accounts already allow validators to set a commission rate for inflation
+rewards and so it's not unreasonable to expect that this commission rate could
+also be used to distribute block rewards. However, some validators have
+expressed a desire to be able to tune revenue streams indpendently.
+
+## New Terminology
+
+- Block Fees Collector: The account used to collect commissioned block fees for
+validators. Previously collected by default into the validator identity account.
+
+- Block Tips Collector: The account used to collect commissioned block tips for
+validators. Previously configured by a Jito CLI parameter.
+
+- Inflation Rewards Collector: The account used to collect commissioned
+inflation rewards for validators. Previously collected by default into the vote
+account.
+
+- Block Fees Commission: The commission rate that determines how much of block
+base fee and priority fee revenue is collected by the validator before
+distributing remaining funds to stake delegators. Previously 100% of block fees
+were distributed to the validator identity account.
+
+- Block Tips Commission: The commission rate that determines how much of block
+tip revenue is collected by the validator before distributing remaining funds to
+stake delegators. Previously configured by a Jito CLI parameter.
+
+- Inflation Rewards Commission: The commission rate that determines how much of 
+stake inflation rewards are collected by the validator before distributing the
+remaining rewards to stake delegators. Previously referred to as simply the
+"commission" since there was no need to differentiate from other types of
+commission.
+
+## Detailed Design
+
+Currently, all block fees, including both transaction base fees and priority
+fees, are collected into a validator's node id account. This proposal details
+changes to the Vote Program and Vote Account that will allow validators to
+specify how different sources of income are collected. This proposal also
+updates the bookkeeping for authorized voters in the vote account state.
+
+### Vote Account 
+
+A new version of vote state will be introduced with the enum discriminant value
+of `3u32` which will be little endian encoded in the first 4 bytes of account
+data. 
+
+```rust
+pub enum VoteStateVersions {
+    V1(..),
+    V2(..),
+    V3(..),
+    V4(..), // <- new variant
+}
+```
+
+This new version of vote state will include new fields for setting the
+commission and collector account for each of the three sources of validator
+income: inflation rewards, block fees, and block tips. It will also remove
+the `prior_voters` field.
+
+```rust
+pub struct VoteStateV4 {
+    pub node_pubkey: Pubkey,
+    pub authorized_withdrawer: Pubkey,
+
+    /// NEW: the collector accounts for validator income
+    pub inflation_rewards_collector: Pubkey,
+    pub block_fees_collector: Pubkey,
+    pub block_tips_collector: Pubkey,
+
+    /// NEW: percentages (0-100) that represent how much of each income source
```

Comment:
nice to have, but not required: bps

Reply:
Yeah I considered that but it seems like even jito mev tip commissions are usually just integer percentages?

Reply:
When dealing with money I think best practice to use bps:

https://github.com/jito-foundation/jito-programs/blob/master/mev-programs/programs/tip-distribution/src/state.rs#L47